### PR TITLE
[SPARK-27141][YARN] Use ConfigEntry for hardcoded configs for Yarn

### PR DESCRIPTION
**Pull request by wangjiaochun**
## What changes were proposed in this pull request?
There is some hardcode configs in code, I think it best to modify。

## How was this patch tested?
Existing tests

### CODE REVIEW

**Comment by vanzin** [comment]
Please clean up the template from the PR description.

**Comment by vanzin** [comment]
ok to test

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala [hunk 1]
```diff
@@ -48,8 +49,8 @@ class MockResolver extends SparkRackResolver {
 class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfterEach {
   val conf = new YarnConfiguration()
   val sparkConf = new SparkConf()
-  sparkConf.set("spark.driver.host", "localhost")
-  sparkConf.set("spark.driver.port", "4040")
+  sparkConf.set(DRIVER_HOST_ADDRESS.key, "localhost")
+  sparkConf.set(DRIVER_PORT.key, "4040")
```
Comment (attilapiros) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: `SparkConf#set` is an overloaded method if you pass a `ConfigEntry` as a first argument it will be typesafe (in this case for `DRIVER_PORT` an Int is expected as second argument).

If the first arg of `SparkConf#set` is a String then the value must be passed as String (not typesafe).

I would prefer to use the typesafe setter and for this only the `.key` method calls should be removed from the ConfigEntries you are using at the calls of `SparkConf#set`.

For the launcher (like in BaseYarnClusterSuite.scala) you can only pass String key when `setConf` is called.

**Comment by attilapiros** [comment]
Regarding the scala style failure you can check the whole project locally with `./dev/sbt-checkstyle`. 

Moreover I strongly recommend to run the tests you touched in that case the build triggers style checking (i.e. using `testOnly` in sbt). For details check [useful developer tools](https://spark.apache.org/developer-tools.html).

**Review by srowen — COMMENTED** [review]
Looks good except yes it should set using type-safe APIs, not the ones taking a string .key

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala (hunk 1 above)
Comment (wangjiaochun) [reply resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: Thank you for your suggestion. I will fix it。

**Comment by wangjiaochun** [comment]
> Regarding the scala style failure you can check the whole project locally with `./dev/sbt-checkstyle`.
> 
> Moreover I strongly recommend to run the tests you touched (i.e. using `testOnly` in sbt) in that case style checking is triggered automatically by the build. For details check [useful developer tools](https://spark.apache.org/developer-tools.html).

I have tried with  ./dev/sbt-checkstyle,but it only return ”Checkstyle checks passed.“

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala [hunk 2]
```diff
@@ -95,9 +96,9 @@ class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfter
       "--class", "SomeClass")
     val sparkConfClone = sparkConf.clone()
     sparkConfClone
-      .set("spark.executor.instances", maxExecutors.toString)
-      .set("spark.executor.cores", "5")
-      .set("spark.executor.memory", "2048")
+      .set(EXECUTOR_INSTANCES.key, maxExecutors.toString)
```
Comment (attilapiros) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: Why not the typesafe method used here (and in the next two lines)?

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala [hunk 3]
```diff
@@ -394,7 +395,7 @@ class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfter
   }
 
   test("window based failure executor counting") {
-    sparkConf.set("spark.yarn.executor.failuresValidityInterval", "100s")
+    sparkConf.set(EXECUTOR_ATTEMPT_FAILURE_VALIDITY_INTERVAL_MS.key, "100s")
```
Comment (attilapiros) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: Same here (remove `.key` transform the value).

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala [hunk 4]
```diff
@@ -394,7 +395,7 @@ class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfter
   }
 
   test("window based failure executor counting") {
-    sparkConf.set("spark.yarn.executor.failuresValidityInterval", "100s")
+    sparkConf.set(EXECUTOR_ATTEMPT_FAILURE_VALIDITY_INTERVAL_MS, 100L)
```
Comment (attilapiros) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: It was `100s` so 100 sec and its unit is milliseconds (because of the suffix _MS) so now `100 * 1000L`

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala [hunk 5]
```diff
@@ -293,8 +293,8 @@ private[spark] class ApplicationMaster(
         Option(appAttemptId.getApplicationId.toString), None).setCurrentContext()
 
       val driverRef = clientRpcEnv.setupEndpointRef(
-        RpcAddress(sparkConf.get("spark.driver.host"),
-          sparkConf.get("spark.driver.port").toInt),
+        RpcAddress(sparkConf.get(DRIVER_HOST_ADDRESS),
+          sparkConf.get(DRIVER_PORT).toInt),
```
Comment (attilapiros) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala]: The `.toInt` is not needed. It is casting and Int to Int.

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala (hunk 4 above)
Comment (vanzin) [reply resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: Actually in this case it's better to set the string directly to "100s" as before.

Comment (attilapiros) [reply resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: I have checked the code after this line and the clock advanced in the same way:
https://github.com/apache/spark/blob/ae205b1d8ee9bc7690205783c2bf1397f492f7b3/resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala#L418-L425

So for me using the same style within a method a bit easier to follow but of course both works.

Comment (attilapiros) [reply resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: @wangjiaochun missing spaces around operator: "100*1000L" => "100 * 1000L"

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala [hunk 6]
```diff
@@ -98,10 +98,10 @@ class YarnClusterSuite extends BaseYarnClusterSuite {
   test("run Spark in yarn-client mode with different configurations, ensuring redaction") {
     testBasicYarnApp(true,
       Map(
-        "spark.driver.memory" -> "512m",
-        "spark.executor.cores" -> "1",
-        "spark.executor.memory" -> "512m",
-        "spark.executor.instances" -> "2",
+         DRIVER_MEMORY.key -> "512m",
```
Comment (srowen) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala]: Nit: remove the extra space indent here and below

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala [hunk 7]
```diff
@@ -394,7 +395,7 @@ class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfter
   }
 
   test("window based failure executor counting") {
-    sparkConf.set("spark.yarn.executor.failuresValidityInterval", "100s")
+    sparkConf.set(EXECUTOR_ATTEMPT_FAILURE_VALIDITY_INTERVAL_MS, 100*1000L)
```
Comment (srowen) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala]: Yes, I'd revert the change to the property value

**Review by srowen — CHANGES_REQUESTED** [review]

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master
